web/packages/teleport/src/UnifiedResources/ResourceCard.tsx

```diff
@@ -121,22 +155,45 @@ export function ResourceCard({ resource, onLabelClick }: Props) {
         showAllLabels={showAllLabels}
         onMouseLeave={() => setShowAllLabels(false)}
       >
-        <ResourceIcon
-          name={resIcon}
-          width="45px"
-          height="45px"
-          ml={2}
-          // We would love to just vertical-center-align this one, but then it
-          // would move down along with expanding the labels. So we apply a
-          // carefully measured top margin instead.
-          mt="16px"
-        />
+        <ResourceIcon name={resIcon} width="45px" height="45px" ml={2} />
         {/* MinWidth is important to prevent descriptions from overflowing. */}
         <Flex flexDirection="column" flex="1" minWidth="0" ml={3} gap={1}>
-          <Flex flexDirection="row" alignItems="start">
-            <SingleLineBox flex="1" title={name}>
-              <Text typography="h5">{name}</Text>
+          <Flex flexDirection="row" alignItems="center">
+            <SingleLineBox flex="1">
+              {isNameOverflowed ? (
+                <HoverTooltip tipContent={<>{name}</>}>
+                  <Text ref={nameTextRef} typography="h5" fontWeight={300}>
+                    {name}
+                  </Text>
+                </HoverTooltip>
+              ) : (
+                <Text ref={nameTextRef} typography="h5" fontWeight={300}>
+                  {name}
+                </Text>
+              )}
             </SingleLineBox>
+            <Popover
+              open={copied}
+              anchorEl={copyAnchorEl.current}
+              anchorOrigin={{
+                vertical: 'top',
+                horizontal: 'center',
+              }}
+              transformOrigin={{
+                vertical: 'bottom',
+                horizontal: 'center',
+              }}
+            >
+              <CopiedMessage>Copied!</CopiedMessage>
```

Comment:
Instead of a "copied!" message, what we do for our `TextSelectCopyMulti` component is change the copy icon to a checkmark icon temporarily. This looks great too though so it's up to you.

Reply:
Hmm, I like that. Also, I need to add a tooltip on hover to even show what this button does so I'll do your suggestion AND keep the copied hover. Thanks!